### PR TITLE
SDWebImagePrefetcher: remove logging

### PR DESCRIPTION
**Pull request by arielelkin**
Remove macro that checks for DEBUG, SD_VERBOSE, and SD_LOG_NONE
Remove calls to NSLog

### CODE REVIEW

**Comment by arielelkin** [comment]
Resubmitting this again, as I believe this addresses worthwhile issues. 

* Consistency is a reason to remove this "feature": `SDWebImagePrefetcher` is the _only_ class in `SDWebImage` that logs to the console. Why should this _completely auxiliary_ class log anything to the console if core classes, such as, `SDImageCache`, `SDWebImageDownloader`, or `UIImageView+WebCache` don't log anything to the console? 

* Lack of control for this logging is another reason for this pull request. The only way to control `SDWebImagePrefetcher`'s logging is by setting its C preprocessor macros, but that can't be done from a Swift project. Swift can _read_ C preprocessor macros, but cannot _set_ them. See the [**Preprocessor Directives**](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/BuildingCocoaApps/InteractingWithCAPIs.html) section of the documentation. So Swift users _cannot_ control this logging. This "feature" is effectively **not compatible** with Swift.


Basically, `SDWebImagePrefetcher` is doing unnecessary and inconsistent logging that can't be controlled with Swift.

**Comment by ElfSundae** [comment]
+1 #1253

**Comment by arielelkin** [comment]
@bpoplauschi 
@mythodeia 
@harishkashyap 

We've been waiting to hear back from you about concrete objections against this pull request.. Let us know if there are any, otherwise, could you please consider merging it? 

Cheers

**Comment by bpoplauschi** [comment]
I agree with the reasons @arielelkin described. Ideally, we would have these logs but since our Swift users will have an issue, I think merging this request makes sense.
